### PR TITLE
DoF: add many user settings

### PR DESCRIPTION
**Pull request by pixelflinger**
- native/half resolution
- gather kernel ring counts
- circle-of-confusion radius clamps

### CODE REVIEW

**Review by romainguy — APPROVED** [review]

File: web/filament-js/jsbindings.cpp [hunk 1]
```diff
@@ -369,6 +369,12 @@ value_object<filament::View::DepthOfFieldOptions>("View$DepthOfFieldOptions")
     .field("maxApertureDiameter", &filament::View::DepthOfFieldOptions::maxApertureDiameter)
     .field("enabled", &filament::View::DepthOfFieldOptions::enabled)
     .field("filter", &filament::View::DepthOfFieldOptions::filter);
+    .field("nativeResolution", &filament::View::DepthOfFieldOptions::nativeResolution);
+    .field("foregroundRingCount", &filament::View::DepthOfFieldOptions::foregroundRingCount);
+    .field("backgroundRingCount", &filament::View::DepthOfFieldOptions::backgroundRingCount);
+    .field("fastGatherRingCount", &filament::View::DepthOfFieldOptions::fastGatherRingCount);
+    .field("maxForegroundCOC", &filament::View::DepthOfFieldOptions::maxForegroundCOC);
+    .field("maxBackgroundCOC", &filament::View::DepthOfFieldOptions::maxBackgroundCOC);
```
Comment (prideout) [inline web/filament-js/jsbindings.cpp:377]: This won't build, it is a daisy chain so does not need semicolons.

**Review by prideout — COMMENTED** [review]
Don't forget `filament.d.ts`

File: filament/CMakeLists.txt [hunk 2]
```diff
@@ -191,6 +191,11 @@ list(APPEND SRCS ${RESGEN_SOURCE})
 # Configuration
 # ==================================================================================================
 
+# whether we're building for mobile -- this can affect some default quality settings
+if (IS_MOBILE_TARGET)
+    add_definitions(-DTARGET_MOBILE=1)
```
Comment (bejado) [inline filament/CMakeLists.txt:196]: Can you preface this definition with `FILAMENT_`?

**Review by bejado — APPROVED** [review]

File: filament/CMakeLists.txt (hunk 2 above)
Comment (pixelflinger) [reply filament/CMakeLists.txt:196]: good idea, I went with the same define we use in shaders, but really this one should've had the prefix too.

**Pull request merged**